pkg/kv/kvtest/store.go

```diff
@@ -462,12 +462,9 @@ func testStoreMissingArgument(t *testing.T, ms MakeStore) {
 	t.Run("Scan", func(t *testing.T) {
 		start := uniqueKey("test-missing-argument")
 		for _, pk := range [][]byte{nil, {}} {
-			it, err := store.Scan(ctx, pk, start)
+			_, err := store.Scan(ctx, pk, start)
 			if !errors.Is(err, kv.ErrMissingPartitionKey) {
-				t.Errorf("Scan using empty partition key - err=%v, expected %s", err, kv.ErrMissingPartitionKey)
-			}
-			if it != nil {
```

Comment:
Why did we remove this check?

Reply:
- Didn't want to check interface value with nil
- Assumed we test only valid protocol - in this case, when we call the function and there is an error, we can't assume anything about the iterator value.